connector/spanmetricsconnector/connector.go

```diff
@@ -306,20 +309,50 @@ func (p *connectorImp) aggregateMetrics(traces ptrace.Traces) {
 				}
 
 				// aggregate histogram metrics
-				h := p.histograms.GetOrCreate(key, attributes)
+				h := histograms.GetOrCreate(key, attributes)
 				h.Observe(latency)
 				if !span.TraceID().IsEmpty() {
 					h.AddExemplar(span.TraceID(), span.SpanID(), latency)
 				}
 
 				// aggregate sums metrics
-				s := p.sums.GetOrCreate(key, attributes)
+				s := sums.GetOrCreate(key, attributes)
 				s.Add(1)
 			}
 		}
 	}
 }
 
+func (p *connectorImp) getOrCreateResourceMetrics(attr pcommon.Map) (metrics.HistogramMetrics, metrics.SumMetrics) {
+	p.keyBuf.Reset()
+	attr.Range(func(k string, v pcommon.Value) bool {
+		p.keyBuf.WriteString(k)
+		p.keyBuf.WriteString(metricKeySeparator)
+		p.keyBuf.WriteString(v.AsString())
+		p.keyBuf.WriteString(metricKeySeparator)
+
+		return true
+	})
+	key := p.keyBuf.String()
```

Comment:
I recommend using [`pdatautil.MapHash`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/73845f4592ebac22a3a9d2e5e8e9e61e7a099eb0/pkg/pdatautil/hash.go#L68). This package has been hardened by some edge cases you may be missing here.